src/elements/EditableDataInputsRow/EditableDataInputsRow.utils.js

```diff
@@ -0,0 +1,123 @@
+import { isNil } from 'lodash'
+
+import { isEveryObjectValueEmpty } from '../../utils/isEveryObjectValueEmpty'
+import { inputsActions } from '../../components/JobsPanelDataInputs/jobsPanelDataInputsReducer'
+import {
+  AZURE_STORAGE_INPUT_PATH_SCHEME,
+  GOOGLE_STORAGE_INPUT_PATH_SCHEME,
+  HTTP_STORAGE_INPUT_PATH_SCHEME,
+  HTTPS_STORAGE_INPUT_PATH_SCHEME,
+  MLRUN_STORAGE_INPUT_PATH_SCHEME,
+  S3_INPUT_PATH_SCHEME,
+  V3IO_INPUT_PATH_SCHEME
+} from '../../constants'
+
+export const applyEditButtonHandler = (
+  handleEdit,
+  inputName,
+  inputsDispatch,
+  requiredField,
+  selectedDataInput,
+  setRequiredField
+) => {
+  if (
+    [
+      HTTP_STORAGE_INPUT_PATH_SCHEME,
+      HTTPS_STORAGE_INPUT_PATH_SCHEME,
+      MLRUN_STORAGE_INPUT_PATH_SCHEME
+    ].includes(selectedDataInput.data.path.pathType)
+  ) {
+    if (
+      !selectedDataInput.data.path.value.split('/')[1] ||
+      selectedDataInput.data.path.value.split('/')[1]?.length === 0
+    ) {
+      return setRequiredField(state => ({ ...state, path: true }))
+    }
+  } else if (selectedDataInput.data.path.url.length === 0) {
+    return setRequiredField(state => ({ ...state, path: true }))
+  }
+
+  if (inputName.length === 0) {
+    return setRequiredField(state => ({ ...state, name: true }))
+  }
+
+  if (!isEveryObjectValueEmpty(requiredField)) {
+    setRequiredField({
+      name: inputName.length > 0,
+      path: !!(
+        selectedDataInput.data.path.value.split('/')[1]?.length > 0 ||
+        selectedDataInput.data.path.url.length > 0
+      )
```

Comment:
The “double bang” operator (`!!`) is unneeded here.
The reason for using it is to convert any JS value to a boolean value, either `true` or `false`.
The `<` operator always gives a boolean value: `value1 < value2` evaluates to either `true` or `false`.
The `||` operator gives either the first or second operand: `value1 || value2` evaluates to either `value1` or `value2`.
Hence:
`value1 > value2 || value3 > value4` evaluates to `true/false || true/false` which evaluates to `true/false`.
Since it give only `true` or `false`, no need for `!!` because no conversion to boolean is needed.
```suggestion
      path:
        selectedDataInput.data.path.value.split('/')[1]?.length > 0 ||
        selectedDataInput.data.path.url.length > 0
```

___

P.S.

Even in cases where a conversion to boolean _is_ needed, I prefer to use the `Boolean` function, because it is more explicit. When the reader sees `Boolean(value)` it is much clearer to them that `value` value will be converted to a boolean value, either `true` or `false`. `!!value` might be less intuitive.